webapp_settings/production.py

```diff
@@ -64,8 +58,6 @@
 
 sentry_sdk.init(
     dsn="{{ vault_sentry_dsn }}",
-    integrations=[DjangoIntegration()]
+    integrations=[DjangoIntegration()],
+    environment=os.environ.get("ENVIRONMENT"),
```

Comment:
So...

I'm afraid it's not quite as simple as this.

First off, the environment would need to be added to the gunicorn systemd script, otherwise the env var will never be set.

However, even once set, we can't tell if we're on a stage or prod environment from the files written to disk - this is because we bake the server images in advance of the deployment, and any image could be deployed to a stage or a prod environment. 

This means we need to set something on the EC2 instance, not on the server image. 

We might be able to do this using the ec2-tag-conditional library I wrote here:

https://github.com/DemocracyClub/ec2-tag-conditional

I suggest we can do this by writing a file to a directory on reboot called `is_prod` or `is_stage`:

```
instance-tags "Env=prod" && touch /is_prod
instance-tags "Env=stage" && touch /is_stage
```

And then adapting the Django settings to see what file exists.

Reply:
I did think this would probably be _too straightforward_ to work. My unfamiliarity with ansible and the deploy process certainly showing here. That library is interesting and seems like it would work though. I've changed this to a draft PR in the meantime

Reply:
You can use ec2-tag-conditional to inspect the tags when bootstrapping. Here's how we do it on WDIV:

https://github.com/DemocracyClub/polling_deploy/blob/9d3dfb1bb2caea7969bef71541e4321bdfe2f936/webapp_settings/production.py#L2-L18

https://github.com/DemocracyClub/polling_deploy/blob/9d3dfb1bb2caea7969bef71541e4321bdfe2f936/webapp_settings/production.py#L79-L85